Bio/Seq.py

```diff
@@ -1700,16 +1705,23 @@ class MutableSeq:
     """
 
     def __init__(self, data):
-        """Initialize the class."""
-        if isinstance(data, str):  # TODO - What about unicode?
+        """Create a MutableSeq object."""
+        if isinstance(data, array.array):
+            if data.typecode != "u":
+                raise ValueError(
+                    "data should be a string, array of characters, Seq object, "
+                    "or MutableSeq object"
+                )
+            self._data = data
+        elif isinstance(data, str):  # TODO - What about unicode?
             self._data = array.array("u", data)
-        elif isinstance(data, (Seq, int, float)):
+        elif isinstance(data, (Seq, MutableSeq)):
```

Comment:
Can't you use a copy of the input MutableSeq's array directly? It seems wasteful to call str on it.